pkg/ast/spl/spl.peg

```diff
@@ -3255,16 +3345,65 @@ FieldNameList <- first:FieldName rest:(COMMA FieldName)* {
 
 TimeModifiers <- EarliestAndLatest / EarliestOnly
 
-EarliestAndLatest <- CMD_EARLIEST EQUAL earliestTime:AbsoluteTimestamp SPACE CMD_LATEST EQUAL latestTime:AbsoluteTimestamp !. {
-    startEpoch, err := ConvertCustomDateTimeFormatToEpochMs(earliestTime.(string))
-    if err != nil {
-        return nil, err
+EarliestAndLatest <- CMD_EARLIEST EQUAL earliestTime:(AbsoluteTimestamp / RelativeTimestamp) SPACE CMD_LATEST EQUAL latestTime:(AbsoluteTimestamp / RelativeTimestamp) !. {
+     var startEpoch int64 = 0
+     var endEpoch int64 = 0
+    var err error
+
+    earliestTimeModifier := earliestTime.(ast.RelativeTimeModifier)
+    latestTimeModifier := latestTime.(ast.RelativeTimeModifier)
+
+    currTime := time.Now()
+    if earliestTimeModifier.AbsoluteTime != "" {
+        startEpoch, err = ConvertCustomDateTimeFormatToEpochMs(earliestTimeModifier.AbsoluteTime)
+        if err != nil {
+            return nil, err
+        }
+    } else {
+        offsetNum := earliestTimeModifier.RelativeTimeOffset.Offset
+        offsetTimeUnit := earliestTimeModifier.RelativeTimeOffset.TimeUnit
+        snap := earliestTimeModifier.Snap
+        t := currTime
+        if offsetNum != 0 {
+            t, err = applyOffset(offsetNum, offsetTimeUnit, t)
+            if err != nil {
+                return nil, err
+            }
+        }
+        if snap != "" {
+            t, err = applySnap(snap, t)
+            if err != nil {
```

Comment:
Can both `snap` and `offsetNum` be given at the same time in the query? If yes, then you are overwriting the `offsetNum` value with `snap` value.

Reply:
From the below Parsing definition at #L3500:
`RelativeTimestamp <- offset:(Offset)? snapParam:(Snap)? { ... }`
I believe both of them can be set. 

If that is not the case, then you need to change the parser to reflect that.

Reply:
First offset is applied and then snap is applied when both are there. If offset is not there then snap is only applied and similar is the case if offset is there and snap is not.